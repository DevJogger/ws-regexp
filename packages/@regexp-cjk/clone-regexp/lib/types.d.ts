import { Options } from 'clone-regexp';
import { IOptionsRewriteFlags } from '@regexp-cjk/rewrite-flags';
export interface ICloneRegexp<T extends RegExp = RegExp> {
    (inputRegExp: T | RegExp, options?: ICloneRegexpOptionsCore, ...argv: any[]): T;
}
export interface ICloneRegexpOptionsCore extends Options, IOptionsRewriteFlags {
}
export interface ICloneRegexpOptions<T extends RegExp = RegExp> extends ICloneRegexpOptionsCore {
    /**
     * allow change cloneRegexp function
     */
    cloneRegexp?: ICloneRegexp<T>;
    disableDetectRegexpClone?: boolean;
}
export declare type IRegExpWithClone<T extends RegExp = RegExp> = T extends {
    clone?(...argv: any[]): any;
} ? T : T & {
    clone?(options?: ICloneRegexpOptionsCore, ...argv: any[]): T;
};
