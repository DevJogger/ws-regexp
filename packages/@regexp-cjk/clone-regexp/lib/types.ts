import { Options } from 'clone-regexp';
import { IOptionsRewriteFlags } from '@regexp-cjk/rewrite-flags';

export interface ICloneRegexp<T extends RegExp = RegExp>
{
	(inputRegExp: T | RegExp, options?: ICloneRegexpOptionsCore, ...argv): T
}

export interface ICloneRegexpOptionsCore extends Options, IOptionsRewriteFlags
{

}

export interface ICloneRegexpOptions<T extends RegExp = RegExp> extends ICloneRegexpOptionsCore
{
	/**
	 * allow change cloneRegexp function
	 */
	cloneRegexp?: ICloneRegexp<T>,
}

export type IRegExpWithClone<T extends RegExp = RegExp> = T extends {
	clone?(...argv)
} ? T : T & {
	clone?(options?: ICloneRegexpOptionsCore, ...argv): T
}
