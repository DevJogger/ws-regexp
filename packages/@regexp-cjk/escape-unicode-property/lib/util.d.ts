import { IOptionsRewritePattern } from '@regexp-cjk/rewrite-pattern';
export declare function handleOptions(options: IOptionsRewritePattern, flags?: string): {
    options: IOptionsRewritePattern;
    flags: string;
};
