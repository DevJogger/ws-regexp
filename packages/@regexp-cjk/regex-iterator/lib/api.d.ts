import { each } from './api/each';
import { map } from './api/map';
import { reduce } from './api/reduce';
import { replaceEach } from './api/replaceEach';
import { splitLimit } from './api/splitLimit';
export { each };
export { map };
export { reduce };
export { replaceEach };
export { splitLimit };
