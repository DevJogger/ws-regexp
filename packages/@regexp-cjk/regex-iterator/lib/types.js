"use strict";
/// <reference lib="es2018.regexp" />
Object.defineProperty(exports, "__esModule", { value: true });
//# sourceMappingURL=types.js.map