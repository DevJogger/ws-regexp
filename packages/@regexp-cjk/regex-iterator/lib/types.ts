/// <reference lib="es2018.regexp" />

declare global
{
	interface RegExpConstructor
	{
		leftContext: string,
		rightContext: string,
	}
}

export interface IReturnTypeEachCore<T extends RegExp = RegExp>
{
	match: RegExpMatchArray;
	re: T;
	index: number;
	data: Pick<typeof RegExp, 'leftContext' | 'rightContext'>;
}

export type ICallback<R = void, T extends RegExp = RegExp> = (match: RegExpMatchArray, index: number, re: T, _: IReturnTypeEachCore<T>) => R

export interface IChainInputObject
{
	/**
	 * The `RegExp` to use.
	 */
	regexp: RegExp;
	/**
	 * The specific backreference.
	 */
	backref?: number | string;

	replaceValue?: string | ((substring: string, ...args: string[]) => string),

}

export type IChainInput = IChainInputObject | RegExp;

export type IChainArray = IChainInput[];

export type IChainInputArray = [RegExp, string | ((substring: string, ...args: string[]) => string)]

