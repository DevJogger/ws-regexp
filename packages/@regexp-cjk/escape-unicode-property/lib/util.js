"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.handleOptions = void 0;
const util_1 = require("@regexp-cjk/rewrite-pattern/lib/util");
const lodash_1 = require("lodash");
function handleOptions(options, flags) {
    return (0, util_1.handleOptions)((0, lodash_1.merge)({}, options, {
        unicodePropertyEscape: true,
        rewriteFlags: {
            unicode: true,
        },
    }), flags !== null && flags !== void 0 ? flags : 'u');
}
exports.handleOptions = handleOptions;
//# sourceMappingURL=util.js.map