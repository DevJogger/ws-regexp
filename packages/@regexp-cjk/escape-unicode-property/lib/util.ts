import { IOptionsRewritePattern } from '@regexp-cjk/rewrite-pattern';
import { handleOptions as _handleOptions } from '@regexp-cjk/rewrite-pattern/lib/util';
import { merge } from 'lodash';

export function handleOptions(options: IOptionsRewritePattern, flags?: string)
{
	return _handleOptions(merge({}, options, {
		unicodePropertyEscape: true,
		rewriteFlags: {
			unicode: true,
		},
	}), flags ?? 'u');
}
