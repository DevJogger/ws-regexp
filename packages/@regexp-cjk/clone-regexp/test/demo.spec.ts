import cloneRegexp from '../index';
import { IRegExpWithClone } from '../lib/types';

test(`default`, () =>
{
	let old = /./i;

	let actual = cloneRegexp(old, {
		global: true,
		ignoreCase: false,
	});

	expect(actual).toHaveProperty('source', old.source);

	expect(actual).toHaveProperty('flags', 'g');

	expect(actual).toMatchSnapshot();

});

test(`custom cloneRegexp`, () =>
{
	let old = /./i;

	let actual = cloneRegexp(old, {
		global: true,
		ignoreCase: false,
		cloneRegexp(re, options)
		{
			return /kkk/
		},
	});

	expect(actual).toHaveProperty('source', 'kkk');

	expect(actual).toHaveProperty('flags', '');

	expect(actual).toMatchSnapshot();

});

test(`re.clone`, () =>
{
	let old: IRegExpWithClone = /./i;

	old.clone = () => {
		return /kkk/m
	};

	let actual = cloneRegexp(old, {
		global: true,
		ignoreCase: false,
	});

	expect(actual).toHaveProperty('source', 'kkk');

	expect(actual).toHaveProperty('flags', 'm');

	expect(actual).toMatchSnapshot();

});
