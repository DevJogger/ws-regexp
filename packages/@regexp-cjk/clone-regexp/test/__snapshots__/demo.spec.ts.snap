// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom cloneRegexp 1`] = `/kkk/`;

exports[`default 1`] = `/\\./g`;

exports[`re.clone 1`] = `/kkk/m`;
